pint/event_toas.py

```diff
@@ -0,0 +1,136 @@
+        'TT' for unmodified events (or geocentered), and
+        'TDB' for events barycentered with gtbary
+    + TIMEREF will be
+        'GEOCENTER' for geocentered events,
+        'SOLARSYSTEM' for barycentered,
+        'LOCAL' for unmodified events
+
+    """
+    if timesys == 'TDB':
+        log.info("Building barycentered TOAs")
+        obs, scale = 'Barycenter', 'tdb'
+    elif timeref == 'LOCAL':
+        log.info(
+            'Building spacecraft local TOAs')
+        obs, scale = mission, 'tt'
+    else:
+        log.info("Building geocentered TOAs")
+        obs, scale = 'Geocenter', 'tt'
+
+    return obs, scale
+
+
+def load_event_TOAs(eventname, mission, weights=None):
+    '''
+    Read photon event times out of a FITS file as PINT TOA objects.
+
+    Correctly handles raw event files, or ones processed with axBary to have
+    barycentered  TOAs. Different conditions may apply to different missions.
+    
+    Parameters
+    ----------
+    eventname : str
+        File name of the FITS event list
+    mission : str
+        Name of the mission (e.g. RXTE, XMM)
+    weights : array or None
+        The array has to be of the same size as the event list. Overwrites 
+        possible weight lists from mission-specific FITS files
+    
+    Returns
+    -------
+    toalist : list of TOA objects
+    '''
+    # Load photon times from event file
+    hdulist = pyfits.open(eventname)
+
+    extension = mission_config[mission]["fits_extension"]
+    allow_local = mission_config[mission]['allow_local']
+
+    if hdulist[1].name not in extension.split(','):
+        raise RuntimeError('First table in FITS file' +
+                           'must be {}. Found {}'.format(extension,
+                                                         hdulist[1].name))
+    event_hdr=hdulist[1].header
+    event_dat=hdulist[1].data
+
+    timesys = event_hdr['TIMESYS']
+    log.info("TIMESYS {0}".format(timesys))
+    if timesys not in ['TDB', 'TT']:
+        raise ValueError('Timesys has to be TDB or TT')
+
+    timeref = event_hdr['TIMEREF']
+    log.info("TIMEREF {0}".format(timeref))
+    if timeref not in ['GEOCENTER', 'SOLARSYSTEM', 'LOCAL']:
+        raise ValueError('Timeref is invalid')
+
+    if allow_local is False and timesys != 'TDB':
+        log.error('Raw spacecraft TOAs not yet supported for ' + mission)
+    # Read time column from FITS file
+    mjds = read_fits_event_mjds_tuples(hdulist[1])
+
+    new_kwargs = {}
+    # Parse and retrieve default values from the FITS columns listed in config
+    cols = mission_config[mission]["fits_columns"]
+    for col in cols.keys():
+        try:
+            val = event_dat.field(cols[col])
+        except ValueError:
+            val = np.zeros(len(mjds))
+        new_kwargs[col] = val
+
+    if weights is not None:
+        new_kwargs["weights"] = weights
+
+    hdulist.close()
+
+    obs, scale = _default_obs_and_scale(mission, timesys, timeref)
+
+    toalist = [None] * len(mjds)
```

Comment:
I'm not sure it applies to this particular case, where the TOA objects are not standard Python variables, but since we are making this array grow it might be more efficient to specify how many entries it will have. I was planning to benchmark it before merging, anyway.